bots/tests-scan

```diff
@@ -78,17 +78,19 @@ def main():
                         help='Display human readable output rather than tasks')
     parser.add_argument('-d', '--dry', action="store_true", default=False,
                         help='Don''t actually change anything on GitHub')
+    parser.add_argument('--repo', default=None,
+                        help='Repository to scan and checkout.')
     parser.add_argument('-c', '--context', action="append", default=[ ],
                         help='Test contexts to use.')
     opts = parser.parse_args()
-    api = github.GitHub()
+    api = github.GitHub(repo=opts.repo)
 
     try:
         if opts.context:
             policy = { c: [ "master" ] for c in opts.context }
         else:
             policy = default_policy()
-        results = scan_for_pull_tasks(api, not opts.dry, opts.human_readable, policy)
+        results = scan_for_pull_tasks(api, not opts.dry, opts.human_readable, policy, opts)
```

Comment:
This is a bit strange - you pass *both* some particular values in `opts` and then the whole `opts` namespace again. AFAICS this is just passing through the new `opts.repo` value, so perhaps make this explicit, to avoid hardcoding assumptions about the `opts` structure into `tests_invoke()`?

Reply:
I did that for the benefit of #8576, where another two parameters would be needed.  What about only passing `opts`?

Reply:
works for me.

Reply:
Done.